### PR TITLE
config/notify: Add SMTP Require TLS global option

### LINKED ISSUE #433: Consider a global config option for require_tls
My setup is to use a relay host to send email alert. This bug might vary depending on the user's need.

**config.yml**
global:
  # The smarthost and SMTP sender used for mail notifications.
  smtp_smarthost: "relayhosthere:25"
  smtp_from: "myname@mydomain.com"
route:
  . . . . . . . . . . .

**Postfix:**

`alias_database = hash:/etc/aliases
alias_maps = hash:/etc/aliases
command_directory = /usr/sbin
config_directory = /etc/postfix
daemon_directory = /usr/libexec/postfix
data_directory = /var/lib/postfix
debug_peer_level = 2
html_directory = no
inet_interfaces = localhost
inet_protocols = all
mail_owner = postfix
mailq_path = /usr/bin/mailq.postfix
manpage_directory = /usr/share/man
mydestination = $myhostname, localhost.$mydomain, localhost
newaliases_path = /usr/bin/newaliases.postfix
queue_directory = /var/spool/postfix
readme_directory = /usr/share/doc/postfix-2.6.6/README_FILES
relayhost = <relay-host-value-here>
sample_directory = /usr/share/doc/postfix-2.6.6/samples
sendmail_path = /usr/sbin/sendmail.postfix
setgid_group = postdrop
unknown_local_recipient_reject_code = 550`

**When I tried to do this setup, I receive an error in the alertmanager log:**

`DEBU[5421] Received alert                                alert=DailyTest[c1fbb74][active] component=dispatcher source=dispatch.go:160
. . . . . . . .
DEBU[5421] notify [DailyTest[c1fbb74][active]]             source=notify.go:400 step=retry
WARN[5421] Notify attempt 1 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5421] Notify attempt 2 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5422] Notify attempt 3 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5422] Notify attempt 4 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5424] Notify attempt 5 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5426] Notify attempt 6 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5429] Notify attempt 7 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5436] Notify attempt 8 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5443] Notify attempt 9 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5453] Notify attempt 10 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5467] Notify attempt 11 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5497] Notify attempt 12 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5544] Notify attempt 13 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5610] Notify attempt 14 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
WARN[5658] Notify attempt 15 failed: require_tls: true (default), but "<relay-host-value-here>" does not advertise the STARTTLS extension  source=notify.go:193
DEBU[5660] Received alert                                alert=DailyTest[c1fbb74][active] component=dispatcher source=dispatch.go:160
ERRO[5721] Error on notify: context deadline exceeded    source=notify.go:152
ERRO[5721] Notify for 1 alerts failed: context deadline exceeded  source=dispatch.go:238
`

I saw a config to turn-off the require_tls by making the value **false** and put it in the global. Then I reload/restart the alertmanager. I still get an error above.

So what I did was to edit the DefaultEmailConfig's RequiredTLS (config/notifiers.go line 36) to false. Compile, start and I was able to receive email alerts.

### PR DESCRIPTION
**Pull request by sgissi**
Fixes #433

### CODE REVIEW

**Comment by sgissi** [comment]
@fabxc forgot to address a dev to review it. Seems like a pretty trivial fix. Go easy on me, it is my first patch here :)

**Comment by brancz** [comment]
LGTM, we should add documentation about this on the Alertmanager configurations docs.

File: notify/impl.go [hunk 1]
```diff
@@ -249,7 +249,8 @@ func (n *Email) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
 		return false, fmt.Errorf("invalid address: %s", err)
 	}
 
-	if n.conf.RequireTLS {
+	// Global Config guarantees RequireTLS is not nil
```
Comment (fabxc) [inline notify/impl.go]: Trailing `.`

**Comment by fabxc** [comment]
LGTM 👍 

Hopefully global options won't get out of hand over time. Then we might want to think about restructuring them.

**Pull request merged**